src/threading.c

```diff
@@ -49,6 +49,24 @@ JL_DLLEXPORT _Atomic(uint8_t) jl_measure_compile_time_enabled = 0;
 JL_DLLEXPORT _Atomic(uint64_t) jl_cumulative_compile_time = 0;
 JL_DLLEXPORT _Atomic(uint64_t) jl_cumulative_recompile_time = 0;
 
+static uint64_t jl_thread_start_time;
+void jl_set_thread_start_time(void)
+{
+    jl_thread_start_time = jl_hrtime();
+}
+
+// TODO: not just current thread
+JL_DLLEXPORT uint64_t jl_thread_up_time(void)
+{
+    return jl_hrtime() - jl_thread_start_time;
+}
+JL_DLLEXPORT uint64_t jl_thread_user_time(void)
+{
+    jl_ptls_t ptls = jl_current_task->ptls;
+    jl_timing_tls_states_t *timing = &ptls->timing_tls;
+    return jl_thread_up_time() - timing->gc_time - timing->lock_spin_time;
```

Comment:
should be 
```suggestion
    return jl_thread_up_time() - timing->gc_time - timing->lock_spin_time - timing->scheduler_time; 
```
but also maybe this isn't the right API, and we should instead have a `jl_thread_timing_stats(int tid)` that populates a struct and do all the arithmetic on the Julia side